### PR TITLE
Switch default smarty for localhost, demo to Smarty5

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Switch default smarty for localhost, demo to Smarty5

Before
----------------------------------------
We were defaulting to Smarty4 for new installs of Civi on demo/ local host sites

After
----------------------------------------
Default is Smarty 5

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: templates/CRM/common/civicrm.settings.php.template [hunk 1]
```diff
@@ -256,18 +256,18 @@ if (!defined('CIVICRM_UF_BASEURL')) {
 /**
  * Specify a Smarty 3 autoload file.
```
Comment (seamuslee001) [inline templates/CRM/common/civicrm.settings.php.template:257]: @eileenmcnaughton should this get updated as well to maybe a Smarty autoload file?

Comment (eileenmcnaughton) [reply templates/CRM/common/civicrm.settings.php.template]: @seamuslee001 yes - good catch

**Comment by seamuslee001** [comment]
MOP

**Comment by demeritcowboy** [comment]
I think we need to figure out the plugins situation. smarty5+civi is still loading the smarty2 plugins dir and deprecates the concept of plugins_dir. It sounds like the preferred way to do custom modifiers is as an [extension](https://smarty-php.github.io/smarty/stable/api/extending/extensions/)

**Comment by eileenmcnaughton** [comment]
@demeritcowboy is that blocking on this PR?

**Comment by demeritcowboy** [comment]
Well, I'm worried things will get lost and I'm not sure what the rush is. This [addition](https://github.com/civicrm/civicrm-packages/commit/31df33086d99e010a4c95aee85db339c2349d72c) added into packages doesn't seem like the right way to do it, and breaks any custom plugin dir. And I still have some concerns about https://github.com/civicrm/civicrm-core/pull/29856/files. It feels like this will get us dug in deeper.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy the reason for putting a class for Smarty5 that would ideally go in vendor was that we realised that if we put it in vendor & it was always-loaded then there were some global namespace functions that would be loaded when older smarty versions were in use. The goal is to move that Smarty class into the base of CiviCRM & the vendor part into the main vendor & always load Smarty5 (only) - but we need to get to the point where it's running OK first - which is why the focus on getting it to be the default on localhost & demo sites now. The rush for me is to get to the point where we are all using Smarty5 in dev-land so we iron out those last things & can get rid of having multiple smarty versions in use.

**Comment by demeritcowboy** [comment]
I'm not trying to block it but for example when you say "The goal is to move that Smarty class into the base of CiviCRM", I'm saying that that smarty class's content does not appear to be the way to do things in smarty 5, and I'm just concerned that will get lost and further entrenched.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy ok - you were concerned about the way plugins are registered here & the cludgey use of require_once

https://github.com/civicrm/civicrm-packages/commit/31df33086d99e010a4c95aee85db339c2349d72c#diff-f3769df2e0f8c8e9d4f2348a2311df0f63d28a1d6493439ba4b5cd4b2af975bdR20

rather than putting non-packages stuff in packages

**Pull request merged**